src/NimBLEUUID.cpp

```diff
@@ -318,4 +320,34 @@ std::string NimBLEUUID::toString() {
     return ble_uuid_to_str(&m_uuid.u, buf);
 } // toString
 
+
+NimBLEUUID & NimBLEUUID::operator=(const NimBLEUUID & other) {
+    m_uuid.u.type = other.m_uuid.u.type;
+    if(other.m_valueSet) {
+        memcpy(m_uuid.u128.value, other.m_uuid.u128.value, 16);
+    }
+    m_valueSet = other.m_valueSet;
+}
+
+bool NimBLEUUID::operator ==(const NimBLEUUID & rhs) {
+    if(m_valueSet && rhs.m_valueSet) {
+        return ble_uuid_cmp(&m_uuid.u, &rhs.m_uuid.u) == 0;
+    }
+
+    return m_valueSet == rhs.m_valueSet;
+}
+
+bool NimBLEUUID::operator !=(const NimBLEUUID & rhs) {
+    return !this->operator==(rhs);
+}
+
+NimBLEUUID::operator std::string() const {
```

Comment:
Not really sure about this one, cast operators are nice, but I don't really see a use case?

Reply:
Well I rather use standard cpp, and would set ::toString() deprecated...

Reply:
Looks good with the change to `:toString()`